openedx/features/enterprise_support/api.py

```diff
@@ -233,52 +259,87 @@ def enterprise_enabled():
     return 'enterprise' in settings.INSTALLED_APPS and getattr(settings, 'ENABLE_ENTERPRISE_INTEGRATION', True)
 
 
-def enterprise_customer_for_request(request, tpa_hint=None):
+def enterprise_customer_for_request(request):
     """
     Check all the context clues of the request to determine if
     the request being made is tied to a particular EnterpriseCustomer.
     """
+
     if not enterprise_enabled():
         return None
 
     ec = None
+    sso_slug = request.GET.get('tpa_hint')
 
     running_pipeline = get_partial_pipeline(request)
     if running_pipeline:
         # Determine if the user is in the middle of a third-party auth pipeline,
-        # and set the tpa_hint parameter to match if so.
-        tpa_hint = Registry.get_from_pipeline(running_pipeline).provider_id
+        # and set the sso_slug parameter to match if so.
```

Comment:
This name should be more consistent; I wanted to switch it to separate the use of an SSO provider's identifier slug in the specific context where it's used to hint to the user how to log in (the tpa_hint query parameter) conceptually from the use of that identifier slug in any other context.